### ISSUE TITLE
Consider adding Cargo.lock back

### ISSUE THREAD

**Issue by tdaede** [xiph/rav1e#1434]
According to https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

cdylibs and binaries should have Cargo.lock, but rust libraries shouldn't. We're all three at once.

**Comment by lu-zero** [comment]
Ideally if we cargo-vendor the sources in the release tarballs `Cargo.lock` is not needed.

**Comment by shssoichiro** [comment]
I'm reasonably certain Cargo will ignore Cargo.lock when rav1e is used as a library, so it should be safe to add back for when it's used as a binary. Is there any reason for specifically wanting it gone?

**Comment by tdaede** [comment]
This page in the rust book specifically says to not include it: https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

**Comment by shssoichiro** [comment]
I'm guessing that's because it won't be used anyway for libraries. I found the section in the Cargo docs that explains it: https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

> Users dependent on the library will not inspect the library’s Cargo.lock (even if it exists).

**Comment by rzumer** [comment]
Moving the binary to its own crate would allow checking in its `Cargo.lock` and not the library's.

**Comment by primeos** [comment]
As a side note: Having an official `Cargo.lock` file available (in this Git repository, release tarball, or e.g. via a separate URL) would also be useful for the [Nix package](https://github.com/NixOS/nixpkgs/pull/75660/files) (vs. checking a custom `Cargo.lock` file with 1340 lines of code into a monorepo for many packages) and [reproducible builds](https://reproducible-builds.org/) in general.

**Comment by lu-zero** [comment]
@Luni-4 can you please look into adding an automation to add the Cargo.lock to the tarball? 

I do hate having the Cargo.lock in the main repository since it makes the diffs fairly large and fairly difficult to read in a normal setup.

**Comment by Luni-4** [comment]
@lu-zero 
Yeah, it would be possible. I will update #1883

**Comment by lu-zero** [comment]
I manually added the Cargo.lock to the release 0.2.0. We could automate this as well.

@primeos would that work for you?

**Comment by Luni-4** [comment]
@lu-zero 

`Cargo.lock` is inside the `rav1e-$version.zip` file

**Comment by primeos** [comment]
@lu-zero yes, that works and the automated solution from @Luni-4 is also fine, thank you very much :) I've merged the Nix(OS) package.

@Luni-4: Maybe it would make sense to add the `Cargo.lock` file separately as it isn't only relevant for Windows users. But I'm also fine with extracting it from the ZIP-Archive. Thank you for automating it.

**Comment by Luni-4** [comment]
> @Luni-4: Maybe it would make sense to add the `Cargo.lock` file separately as it isn't only relevant for Windows users. But I'm also fine with extracting it from the ZIP-Archive. Thank you for automating it.

Yep, you're right. I will change that as soon as possible. Thanks! :)

**Comment by Luni-4** [comment]
@primeos 

Changed in #1973
